commands/audit/audit.go

```diff
@@ -88,16 +95,62 @@ func (auditCmd *AuditCommand) SetThreads(threads int) *AuditCommand {
 	return auditCmd
 }
 
-func (auditCmd *AuditCommand) CreateCommonGraphScanParams() *scangraph.CommonGraphScanParams {
-	commonParams := &scangraph.CommonGraphScanParams{
-		RepoPath: auditCmd.targetRepoPath,
-		Watches:  auditCmd.watches,
-		ScanType: services.Dependency,
+func (auditCmd *AuditCommand) CreateAuditResultsContext(serverDetails *config.ServerDetails) (context results.ResultContext) {
+	return CreateResultsContext(
+		serverDetails,
+		auditCmd.GetXrayVersion(),
+		auditCmd.watches,
+		auditCmd.targetRepoPath,
+		auditCmd.projectKey,
+		auditCmd.gitRepoHttpsCloneUrl,
+		auditCmd.IncludeVulnerabilities,
+		auditCmd.IncludeLicenses,
+	)
+}
+
+// Create a results context based on the provided parameters. resolves conflicts between the parameters based on the retrieved platform watches.
+func CreateResultsContext(serverDetails *config.ServerDetails, xrayVersion string, watches []string, artifactoryRepoPath, projectKey, gitRepoHttpsCloneUrl string, includeVulnerabilities, includeLicenses bool) (context results.ResultContext) {
+	context = results.ResultContext{
+		RepoPath:               artifactoryRepoPath,
+		Watches:                watches,
+		ProjectKey:             projectKey,
+		IncludeVulnerabilities: shouldIncludeVulnerabilities(includeVulnerabilities, watches, artifactoryRepoPath, projectKey, ""),
+		IncludeLicenses:        includeLicenses,
+	}
+	if err := clientutils.ValidateMinimumVersion(clientutils.Xray, xrayVersion, services.MinXrayVersionGitRepoKey); err != nil {
+		// Git repo key is not supported by the Xray version.
+		return
+	}
+	if gitRepoHttpsCloneUrl == "" {
+		// No git repo key was provided, no need to check anything else.
+		log.Debug("Git repo key was not provided, jas violations will not be checked.")
+		return
+	}
+	// Get the defined and active watches from the platform.
+	manager, err := xsc.CreateXscService(serverDetails)
+	if err != nil {
+		log.Warn(fmt.Sprintf("Failed to create Xray services manager: %s", err.Error()))
+		return
+	}
+	if context.PlatformWatches, err = manager.GetResourceWatches(xscutils.GetGitRepoUrlKey(gitRepoHttpsCloneUrl), projectKey); err != nil {
+		log.Warn(fmt.Sprintf("Failed to get active defined watches: %s", err.Error()))
+		return
+	}
+	// Set git repo key and check if it has any watches defined in the platform.
+	context.GitRepoHttpsCloneUrl = gitRepoHttpsCloneUrl
+	if len(context.PlatformWatches.GitRepositoryWatches) == 0 && len(watches) == 0 && projectKey == "" {
+		log.Debug(fmt.Sprintf("Git repo key was provided (%s) but no watches were defined in the platform. ignoring git repo key...", context.GitRepoHttpsCloneUrl))
```

Comment:
I think the log should be something like:
"No watches were found in the platform for the given git repo key (%s), and no watches were given by the user (using watches or project flags). Ignoring git repo and calculating vulnerabilities..."